### DISCUSSION TITLE
`duckdb.connect()` with Python API should accept `Path` objects

### DISCUSSION THREAD

**Discussion by flexatone** [duckdb/duckdb#12778] [Ideas]
### What happens?

With the Python API, `Path` instances are not accepted as file paths. 

### To Reproduce

```python
>>> import duckdb
>>> from pathlib import Path
>>> c1 = duckdb.connect('/tmp/test1.db')
>>> c2 = duckdb.connect(Path('/tmp/test2.db'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: connect(): incompatible function arguments. The following argument types are supported:
    1. (database: str = ':memory:', read_only: bool = False, config: dict = None) -> duckdb.DuckDBPyConnection

Invoked with: PosixPath('/tmp/test2.db')
```

### OS:

MacOS 14.1.1, arm64

### DuckDB Version:

1.0.0

### DuckDB Client:

Python

### Full Name:

Christopher Ariza

### Affiliation:

Self

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

No - Other reason (please specify in the issue body)

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by soerenwolfers** [comment]
This is a feature request, and was previously requested at https://github.com/duckdb/duckdb/discussions/8878
